test/cpp/naming/gen_build_yaml.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python2.7
+# Copyright 2015 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Generates the appropriate build.json data for all the end2end tests."""
+
+
+import yaml
+import collections
+import hashlib
+import json
+
+_LOCAL_DNS_SERVER_ADDRESS = '127.0.0.1:15353'
+
+def _append_zone_name(name, zone_name):
+  return '%s.%s' % (name, zone_name)
+
+def _build_expected_addrs_cmd_arg(expected_addrs):
+  out = []
+  for addr in expected_addrs:
+    out.append('%s,%s' % (addr['address'], str(addr['is_balancer'])))
+  return ';'.join(out)
+
+def main():
+  resolver_component_data = ''
+  with open('test/cpp/naming/resolver_test_record_groups.yaml') as f:
+    resolver_component_data = yaml.load(f)
+
+  json = {
+      'resolver_component_test_cases': [
+          {
+              'target_name': _append_zone_name(test_case['record_to_resolve'],
+                                                 resolver_component_data['resolver_component_tests_common_zone_name']),
+              'expected_addrs': _build_expected_addrs_cmd_arg(test_case['expected_addrs']),
+              'expected_chosen_service_config': (test_case['expected_chosen_service_config'] or ''),
+              'expected_lb_policy': (test_case['expected_lb_policy'] or ''),
+          } for test_case in resolver_component_data['resolver_component_tests']
+      ],
+      'libs': [
+          {
+              'name': 'resolver_component_tests_runner_invoker_common',
+              'build': 'private',
+              'language': 'c++',
+              'src': ['test/cpp/naming/resolver_component_tests_runner_invoker_common.cc'],
+              'headers': ['test/cpp/naming/resolver_component_tests_runner_invoker_common.h'],
+              'platforms': ['linux', 'posix', 'mac'],
```

Comment:
@nicolasnoble, I assume that we'll need to add windows to this list as part of your #12416.  As per our discussion, this PR will be merged before that one.